docs: src/guide/migration/v-on-native-modifier-removed.md (#300)
vuejs/docs-next-zh-cn: src/guide/migration/v-on-native-modifier-removed.md

---
title: 移除 v-on.native 修饰符
badges:
  - breaking
---

# 移除 `v-on.native` 修饰符 <MigrationBadges :badges="$frontmatter.badges" />

## 概览

`v-on` 的 `.native` 修饰符已被移除。

## 2.x 语法

> 默认情况下，传递给组件带有 `v-on` 的事件监听器只有通过 `this.$emit` 才能触发。要将原生 DOM 监听器添加到子组件的根元素中，可以使用 `.native` 修饰符：

Reviewer: ```suggestion
默认情况下，传递给带有 `v-on` 的组件的事件监听器只有通过 `this.$emit` 才能触发。要将原生 DOM 监听器添加到子组件的根元素中，可以使用 `.native` 修饰符：
```

/>
```

## 3.x 语法

> `v-on` 的 `.native` 修饰符已被移除。同时，[新增的 `emits` 选项](./emits-option.md)允许 child 定义真正触发的事件。

Reviewer: ```suggestion
`v-on` 的 `.native` 修饰符已被移除。同时，[新增的 `emits` 选项](./emits-option.md)允许子组件定义真正会被触发的事件。
```

> 因此，对于 child 中*未*定义为组件-触发事件的所有事件监听器，Vue 现在将把他们作为原生事件监听器添加到 child 的根元素中(除非在 child 的选项中设置了 `inheritAttrs: false`)。

Reviewer: ```suggestion
因此，对于子组件中*未*被定义为组件触发的所有事件监听器，Vue 现在将把它们作为原生事件监听器添加到子组件的根元素中 (除非在子组件的选项中设置了 `inheritAttrs: false`)。
```
